### PR TITLE
refactor(medusa): move repository specs into its own folder

### PR DESCRIPTION
**Pull request by riqwan**
**What:**

Introduces a new folder under which repository specs will be placed. 

Why:

We don't currently have a good place to test ORM logic or custom queries against the database. The repository folder tests are a place for just exactly that. 

How:

Creates an internal package similar to other integration tests - api and plugins. 

RESOLVES CORE-965

### CODE REVIEW

File: integration-tests/repositories/factories/simple-product-category-factory.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Connection } from "typeorm"
```
Comment (olivermrbl) [inline integration-tests/repositories/factories/simple-product-category-factory.ts:1]: question: Now that we have started using factories across multiple projects, should we also centralize them in a dedicated place? Wdyt?

**Review by olivermrbl — COMMENTED** [review]
This is great @riqwan! 

Can I get you to add the folder to `ignore` in `.changeset/config.json`?

File: integration-tests/repositories/factories/simple-product-category-factory.ts (hunk 1 above)
Comment (riqwan) [reply integration-tests/repositories/factories/simple-product-category-factory.ts:1]: I'd prefer that. Is there any reason why they're independent packages and not just one?

Either case, I'll create an issue for centralising the factories.

Comment (olivermrbl) [reply integration-tests/repositories/factories/simple-product-category-factory.ts:1]: > Is there any reason why they're independent packages and not just one

I am not sure I follow. What packages?

Comment (riqwan) [reply integration-tests/repositories/factories/simple-product-category-factory.ts:1]: @olivermrbl the `integration-test/(api|plugins|repositories)` folders are created as independent private packages.

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**